### PR TITLE
Implement parsing of ia command zappy server

### PR DESCRIPTION
**Pull request by AxelF44**
I've implemented the parsing and all command of AI !

### CODE REVIEW

File: server/include/ai/cmd/command_ai.h [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+** Zappy Server
+** File description:
+** AI Command
+*/
+
+#pragma once
+
+#include "ai/ai.h"
+
+#define BEGIN 1
```
Comment (Njord201) [inline server/include/ai/cmd/command_ai.h:12]: Update the name of the macro, giving it a more explicit name such as SATUS_BEGIN.

File: server/include/ai/cmd/command_ai.h [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+#define END 2
```
Comment (Njord201) [inline server/include/ai/cmd/command_ai.h:13]: Same

File: server/include/ai/cmd/command_ai.h [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+ * @brief Handler of AI command.
+ *
+ * @param app Application with necessary information.
+ * @param ai AI who have done a command.
+ * @param line The command of AI.
+ */
+void command_ai_handler(app_t *app, ia_t *ai, char *line);
+
+/**
+ * @brief Handler of command who move player(Forward, Right, Left)
```
Comment (Njord201) [inline server/include/ai/cmd/command_ai.h]:
```suggestion
 * @brief Handler of command who move player (Forward, Right, Left).
```

File: server/src/ai/cmd/broadcast/broadcast_cmd.c [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        }
+        temp = temp->next;
+    }
+}
+
+bool broadcast_command(app_t *app, ia_t *ai, char *line)
+{
+    char *reply = NULL;
+
+    if (strncmp("Broadcast", line, 9) == 0) {
```
Comment (Njord201) [inline server/src/ai/cmd/broadcast/broadcast_cmd.c:43]: If the ai send "Broadcastdedzef", how will the programme respond?

File: server/src/ai/cmd/eject/eject_egg.c [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+#include "server/client.h"
+#include "ai/cmd/command_ai.h"
+
+static void check_position_egg(team_t *team,
+    list_node_t *egg_temp, vector2i_t *ai_position)
+{
+    if (egg_temp->data.coord->x == ai_position->x &&
+    egg_temp->data.coord->y == ai_position->y) {
+        list_delete(team->egg_position, egg_temp);
+    }
```
Comment (Njord201) [inline server/src/ai/cmd/eject/eject_egg.c:18]: Remove the brackets

File: server/src/ai/cmd/eject/eject_egg.c [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    list_node_t *next_egg = NULL;
+
+    while (temp != NULL) {
+        team = temp->data.team;
+        egg_temp = team->egg_position->first;
+        while (egg_temp != NULL) {
+            next_egg = egg_temp->next;
+            check_position_egg(team, egg_temp, ai->position);
+            egg_temp = next_egg;
+        }
```
Comment (Njord201) [inline server/src/ai/cmd/eject/eject_egg.c:35]: You don't need next_egg variable or egg_temp.
```suggestion
        while (egg_temp != NULL) {
            check_position_egg(team, egg_temp, ai->position);
            egg_temp = egg_temp->next;
        }
```

File: server/src/ai/cmd/eject/eject_player.c [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+        eject_response_west(ai);
+    }
+}
+
+static void check_position_player(app_t *app, ia_t *ai, ia_t *ai_eject)
+{
+    if (ai->position->x == ai_eject->position->x &&
+    ai->position->y == ai_eject->position->y && ai->fd != ai_eject->fd) {
+        eject_direction(app, ai, ai_eject);
+    }
```
Comment (Njord201) [inline server/src/ai/cmd/eject/eject_player.c:101]: Remove the brackets

File: server/src/ai/cmd/look/look_east.c [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    for (int level_ai = index_line; level_ai != 0; level_ai--) {
+        if (depart_x + 1 == (int) app->game->width)
+            depart_x = 0;
+        else
+            depart_x++;
+    }
+    return depart_x;
+}
+
+static int decremente_value(app_t *app, int depart_y, int decompte)
```
Comment (Njord201) [inline server/src/ai/cmd/look/look_east.c:24]: French variable ??

File: server/src/ai/cmd/look/look_east.c [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+        return 0;
+    else
+        return depart_y + decompte;
+}
+
+static int get_y_line(app_t *app, ia_t *ai, int index_tile)
+{
+    int depart_y = ai->position->y;
+    int goal = index_tile;
+    int decompte = 0;
```
Comment (Njord201) [inline server/src/ai/cmd/look/look_east.c:38]: French variable

File: server/src/ai/cmd/look/look_east.c [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+    if (index_tile == 0)
+        return depart_y;
+    if (index_tile < 0)
+        decompte = -1;
+    else
+        decompte = 1;
+    for (int index = 0; index != goal; index += decompte) {
+        depart_y = decremente_value(app, depart_y, decompte);
+    }
```
Comment (Njord201) [inline server/src/ai/cmd/look/look_east.c]: Remove this brackets

File: server/src/ai/cmd/look/look_handler.c [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+#include "ai/cmd/command_ai.h"
+
+static void concat_player(vector2i_t *pos, list_node_t *ia_temp,
+    char **reply, ia_t *ai)
+{
+    if (ia_temp->data.ai->position->x == pos->x &&
+    ia_temp->data.ai->position->y == pos->y &&
+    ia_temp->data.ai->fd != ai->fd) {
+        concatenate_strings(reply, PLAYER_STRING);
+    }
```
Comment (Njord201) [inline server/src/ai/cmd/look/look_handler.c:19]: Brackets

File: server/src/ai/cmd/look/look_handler.c [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+static void concat_egg(vector2i_t *pos, list_node_t *egg_temp,
+    char **reply)
+{
+    if (egg_temp->data.coord->x == pos->x &&
+    egg_temp->data.coord->y == pos->y) {
+        concatenate_strings(reply, EGG_STRING);
+    }
```
Comment (Njord201) [inline server/src/ai/cmd/look/look_handler.c:45]: Brackets

File: server/src/ai/cmd/look/look_handler.c [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+        concatenate_strings(reply, MENDIANE_STRING);
+    for (size_t i = 0; i < app->game->map[pos->y][pos->x].phiras; i++)
+        concatenate_strings(reply, PHIRAS_STRING);
+    for (size_t i = 0; i < app->game->map[pos->y][pos->x].thystame; i++)
+        concatenate_strings(reply, THYSTAME_STRING);
+}
+
+void look_cmd(app_t *app, ia_t *ai)
+{
+    char *reply = strdup("[player");
```
Comment (Njord201) [inline server/src/ai/cmd/look/look_handler.c:85]: Check your strdup

File: server/src/ai/cmd/look/look_north.c
Comment (Njord201) [inline server/src/ai/cmd/look/look_north.c:1]: Brackets and french variable.

File: server/src/ai/cmd/look/look_south.c
Comment (Njord201) [inline server/src/ai/cmd/look/look_south.c:1]: Brackets and french variable.

File: server/src/ai/cmd/look/look_west.c
Comment (Njord201) [inline server/src/ai/cmd/look/look_west.c:1]: Brackets and french variable.

File: server/src/ai/die.c [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+    change_ai(app, ai);
+    dead_response(app);
+}
+
+static void check_statut_life(app_t *app, ia_t *ai)
+{
+    if (time_elapsed(&ai->time->start_life) >= ai->time->total_life) {
+        if (ai->inventory->food == 0) {
+            ia_die(app, ai);
+        }
```
Comment (Njord201) [inline server/src/ai/die.c:61]: Remove brackets

File: server/src/game/game.c [hunk 15]
```diff
@@ -7,16 +7,100 @@
...
     if (new_game->map == NULL)
         return NULL;
     return new_game;
 }
+
+void destroy_game(game_t *game)
+{
+    for (size_t index_y = 0; index_y < game->height; index_y++) {
+        free(game->map[index_y]);
+    }
```
Comment (Njord201) [inline server/src/game/game.c:103]: Remove brackets

**Review by Njord201 — CHANGES_REQUESTED** [review]
Some requests

**Review by Njord201 — APPROVED** [review]

**Review by mathieurobert1 — APPROVED** [review]
Nice job ! For the future, please be careful with the name of the issue and its content. This pr is so big.

**Pull request merged**